### ISSUE TITLE
PPO training with multi-node

### ISSUE THREAD

**Issue by GGgrey** [OpenRLHF/OpenRLHF#945]
![Image](https://github.com/user-attachments/assets/eb29843c-8ee8-414d-bfbf-54c2edc557be)

Hi,

I am running 8B model on a two nodes with 2 gpus per node, but it always runs stuck at the position shown in the picture. Can you help me see where I'm going wrong?

Here are my actions,
ray start --head --dashboard-host=0.0.0.0 --node-ip-address 10.184.27.13 --num-gpus 2
ray start --address='10.184.27.13:6379' --num-gpus 2

ray job submit --address="http://10.184.27.13:8265" \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 2 \
   --reward_num_nodes 1 \
   --reward_num_gpus_per_node 2 \
   --critic_num_nodes 1 \
   --critic_num_gpus_per_node 2 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 2 \
   --vllm_num_engines 1 \
   --vllm_tensor_parallel_size 2 \
   --colocate_all_models \
   --vllm_gpu_memory_utilization 0.5 \
   --pretrain /models/Llama-3-8b-sft-mixture \
   --reward_pretrain /models/Llama-3-8b-rm-mixture \
   --save_path /openrlhf/examples/test_scripts/final/llama3-8b-rlhf \
   --ckpt_path /openrlhf/examples/test_scripts/ckpt/llama3-8b-rlhf \
   --save_hf_ckpt \
   --micro_train_batch_size 8 \
   --train_batch_size 128 \
   --micro_rollout_batch_size 16 \
   --rollout_batch_size 1024 \
   --n_samples_per_prompt 1 \
   --max_epochs 1 \
   --prompt_max_len 1024 \
   --max_samples 100000 \
   --generate_max_len 1024 \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 5e-7 \
   --critic_learning_rate 9e-6 \
   --init_kl_coef 0.01 \
   --prompt_data /datasets/prompt-collection-v0.1 \
   --input_key context_messages \
   --apply_chat_template \
   --normalize_reward \
   --gradient_checkpointing \
   --adam_offload \
   --flash_attn \
   --packing_samples \
   --vllm_sync_backend nccl \
   --enforce_eager \
   --vllm_enable_sleep \
   --deepspeed_enable_sleep
